tests/test_install_deps.py

```diff
@@ -0,0 +1,80 @@
+"""Verify behavior of TemporaryPipInstallResolver."""
+import logging
+
+import pytest
+
+from fawltydeps.packages import (
+    IdentityMapping,
+    Package,
+    TemporaryPipInstallResolver,
+    resolve_dependencies,
+)
+
+from .project_helpers import TarballPackage
+
+
+@pytest.fixture
+def local_pip_cache(request, monkeypatch):
+    cache_dir = TarballPackage.cache_dir(request.config.cache)
+    TarballPackage.get_tarballs(request.config.cache)
+    # set the test's env variables so that pip would install from the local repo
+    monkeypatch.setenv("PIP_NO_INDEX", "True")
+    monkeypatch.setenv("PIP_FIND_LINKS", str(cache_dir))
+
+
+def test_resolve_dependencies_install_deps__via_local_cache(local_pip_cache):
+    debug_info = "Provided by temporary `pip install`"
+
+    actual = resolve_dependencies(
+        ["leftpadx", "click"], pyenv_path=None, install_deps=True
+    )
+    assert actual == {
+        "leftpadx": Package(
+            "leftpadx", {"leftpad"}, TemporaryPipInstallResolver, debug_info
+        ),
+        "click": Package("click", {"click"}, TemporaryPipInstallResolver, debug_info),
+    }
+
+
+def test_resolve_dependencies_install_deps__handle_pip_install_failure(
+    caplog, local_pip_cache
+):
+    # TemporaryPipInstallResolver will handle the inevitable pip install error
+    # and return to resolve_dependencies() with the missing package unresolved.
+    # For now, IdentityMapping "saves the day" and supplies a Package object.
+    # Soon, this should result in an unresolved package error instead.
+    caplog.set_level(logging.WARNING)
+    actual = resolve_dependencies(
+        ["package_does_not_exist"], pyenv_path=None, install_deps=True
+    )
+    assert actual == {
+        "package_does_not_exist": Package(
+            "package_does_not_exist", {"package_does_not_exist"}, IdentityMapping
+        ),
+    }
+    assert all(
+        word in caplog.text for word in ["pip", "install", "package_does_not_exist"]
+    )
+
+
+def test_resolve_dependencies_install_deps__pip_install_some_packages(
+    caplog, local_pip_cache
+):
+    debug_info = "Provided by temporary `pip install`"
+    caplog.set_level(logging.WARNING)
+    actual = resolve_dependencies(
+        ["leftpadx", "package_does_not_exist"], pyenv_path=None, install_deps=True
```

Comment:
```suggestion
        ["leftpadx", "package_does_not_exist", "click"], pyenv_path=None, install_deps=True
```
Add a legit package after the fake one, to make sure `pip install` did `retry_each`.

Reply:
Agreed. However, `TemporaryPipInstallResolver.lookup_packages()` _sorts_ the package names before passing them to `.temp_installed_requirements()` so in order for this to be properly tested, we need package names that sort alphabetically before and after the missing package Will fix.